make-coe/make-coe_test.go

```diff
@@ -3,42 +3,68 @@ package makecoe
 
 	"github.com/getcarina/carina/common"
 )
 
-func TestDeleteClusterThenErrorOutHaltsWait(t *testing.T) {
-	common.Log.RegisterTestLogger(t)
+const identityAPIVersion = "/v2.0/"
+
+var anyClusterRegexp = regexp.MustCompile("^/clusters/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
+
+type handler func(w http.ResponseWriter, r *http.Request)
+
+func identityHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
 
-	mockCarina := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	switch r.RequestURI {
+	case "/v2.0/tokens":
+		fmt.Fprintln(w, `{"access":{"serviceCatalog":[{"endpoints":[{"tenantId":"963451","publicURL":"https:\/\/api.dfw.getcarina.com","region":"DFW"}],"name":"cloudContainer","type":"rax:container"}],"user":{"name":"fake-user","id":"fake-userid"},"token":{"expires":"3000-01-01T12:00:00Z","id":"fake-token","tenant":{"name":"fake-tenantname","id":"fake-tenantid"}}}}`)
+	default:
+		w.WriteHeader(404)
+		fmt.Fprintln(w, "unexpected request: "+r.RequestURI)
+	}
+}
+
+func clusterInErrorHandler(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "application/json")
 
-		switch r.RequestURI {
-		case "/":
-			fmt.Fprintln(w, `{"versions": [{"status": "current", "min_version": "1.0", "max_version": "1.0", "id": "v1.0"}]}`)
-		case "/clusters/99999999-9999-9999-9999-999999999999":
+		switch {
+		case anyClusterRegexp.MatchString(r.RequestURI):
 			fmt.Fprintln(w, `{ "status": "error" }`)
-		case "/v2.0/tokens":
-			fmt.Fprintln(w, `{"access":{"serviceCatalog":[{"endpoints":[{"tenantId":"963451","publicURL":"https:\/\/api.dfw.getcarina.com","region":"DFW"}],"name":"cloudContainer","type":"rax:container"}],"user":{"name":"fake-user","id":"fake-userid"},"token":{"expires":"3000-01-01T12:00:00Z","id":"fake-token","tenant":{"name":"fake-tenantname","id":"fake-tenantid"}}}}`)
 		default:
 			fmt.Fprintln(w, "unexpected request: "+r.RequestURI)
 			w.WriteHeader(404)
 		}
-	}))
-	defer mockCarina.Close()
+}
 
+func createMockCarina(h handler) (*httptest.Server, *httptest.Server) {
+	return httptest.NewServer(http.HandlerFunc(h)), httptest.NewServer(http.HandlerFunc(identityHandler))
+}
+
+func createMakeCOEService(identity string, endpoint string) *MakeCOE {
 	acct := &Account{
-		AuthEndpointOverride: mockCarina.URL + "/v2.0/",
-		EndpointOverride:     mockCarina.URL,
+		AuthEndpointOverride: identity,
+		EndpointOverride:     endpoint,
 		UserName:             "fake-user",
 		APIKey:               "fake-apikey",
 		Region:               "DFW",
 	}
 
-	svc := &MakeCOE{Account: acct}
+	return &MakeCOE{Account: acct}
+}
+
+func TestDeleteClusterThenErrorOutHaltsWait(t *testing.T) {
+	common.Log.RegisterTestLogger(t)
+
+	mockCarina, mockIdentity := createMockCarina(clusterInErrorHandler)
+	defer mockCarina.Close()
+	defer mockIdentity.Close()
+
+	svc := createMakeCOEService(mockIdentity.URL + identityAPIVersion, mockCarina.URL)
```

Comment:
What if we pass in the mocks servers into `createMakeCOEService` and have that handle in one place appending the identity to the URL?